packages/vm/src/imports.rs

```diff
@@ -98,8 +98,30 @@ pub fn do_db_write<A: BackendApi + 'static, S: Storage + 'static, Q: Querier + '
         return Err(VmError::write_access_denied());
     }
 
-    let key = read_region(&data.memory(&store), key_ptr, MAX_LENGTH_DB_KEY)?;
-    let value = read_region(&data.memory(&store), value_ptr, MAX_LENGTH_DB_VALUE)?;
+    /// Converts a region length error to a different variant for better understandability
+    fn convert_error(e: VmError, kind: &'static str) -> VmError {
+        if let VmError::CommunicationErr {
+            source: CommunicationError::RegionLengthTooBig { length, max_length },
+            #[cfg(feature = "backtraces")]
+            backtrace,
+        } = e
+        {
+            VmError::GenericErr {
+                msg: format!(
+                "{kind} too big. Tried to write {length} bytes to storage, limit is {max_length}"
+            ),
+                #[cfg(feature = "backtraces")]
+                backtrace,
+            }
```

Comment:
I think the backtrace is lost anyways in an import implementation error case. So you can just use `VmError::generic_err(...)` here

Reply:
Oh, I see, those are emitted as traps by wasmer, caught again by us and converted to VmError using the `Display` impl. Then it indeed makes no sense to keep the backtrace.

Reply:
Yeah exactly. Was not 100% sure about the details but this is pretty much it. 

> using the Display impl.

This is why we don't need a new structured error in this case.